### PR TITLE
Commiting Service Now Connector

### PR DESCRIPTION
**Pull request by yashothara**

### CODE REVIEW

File: src/main/assembly/assemble-connector.xml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Kanapriya) [inline src/main/assembly/assemble-connector.xml]: Add license header in this file

File: src/main/assembly/assemble-connector.xml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+<assembly>
+    <formats>
+        <format>zip</format>
+    </formats>
+
```
Comment (Kanapriya) [inline src/main/assembly/assemble-connector.xml]: Remove unwanted lines

File: src/main/assembly/filter.properties [hunk 3]
```diff
@@ -0,0 +1,3 @@
+product.name=WSO2 ESB
+product.version=4.9.0-SNAPSHOT
+
```
Comment (Kanapriya) [inline src/main/assembly/filter.properties]: Remove empty lines

File: src/main/assembly/filter.properties [hunk 4]
```diff
@@ -0,0 +1,3 @@
+product.name=WSO2 ESB
+product.version=4.9.0-SNAPSHOT
```
Comment (Kanapriya) [inline src/main/assembly/filter.properties]: Why SNAPSHOT version is used?

File: src/test/INTEGRATION-TEST.txt [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+	xix)    sysparmInputDisplayValue         -  Use true or false.
+	xx)     sysparmAvgFields                 -  Use valid attribute names of the table.
+	xxi)    sysparmMinFields	         -  Use valid attribute names of the table.
+	xxii)   sysparmMaxFields                 -  Use valid attribute names of the table.
+	xxiii)  sysparmSumFields		 -  Use valid attribute names of the table.
+	xxiv)   sysparmCount		         -  Use true or false.
+	
+ 7. Navigate to "{SERVICENOW_CONNECTOR_HOME}/" and run the following command.
+      $ mvn clean install
+
```
Comment (Kanapriya) [inline src/test/INTEGRATION-TEST.txt]: Remove unwanted lines

File: src/test/resources/artifacts/ESB/config/restRequests/servicenow/patchRecordById_optional.json [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  "approval":"%s(approval)",
+  "category":"%s(category)",
+  "apiColumns": {"close_notes":"close clogard","correlation_display":"api"},
+  "sysparmDisplayValue":"%s(sysparmDisplayValue)",
+  "sysparmFields":"%s(sysparmFields)",
+  "sysparmView":"%s(sysparmView)",
+  "sysparmExcludeReferenceLink":"",
+  "sysparmInputDisplayValue ":"%s(sysparmInputDisplayValue)"
+}
+
```
Comment (Kanapriya) [inline src/test/resources/artifacts/ESB/config/restRequests/servicenow/patchRecordById_optional.json]: Remove unwanted lines

File: src/test/resources/artifacts/ESB/config/restRequests/servicenow/postRecord_negative.json [hunk 7]
```diff
@@ -0,0 +1,7 @@
+{
+  "serviceNowInstanceURL":"%s(serviceNowInstanceURL)",
+  "username":"%s(username)",
+  "password":"%s(password)",
+  "tableName":"Negative"
+}
+
```
Comment (Kanapriya) [inline src/test/resources/artifacts/ESB/config/restRequests/servicenow/postRecord_negative.json]: Remove empty lines

File: src/test/resources/artifacts/ESB/config/restRequests/servicenow/putRecordById_optional.json [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+  "approval":"%s(approval)",
+  "category":"%s(category)",
+  "apiColumns": {"close_notes":"close clogard","correlation_display":"api"},
+  "sysparmDisplayValue":"%s(sysparmDisplayValue)",
+  "sysparmFields":"%s(sysparmFields)",
+  "sysparmView":"%s(sysparmView)",
+  "sysparmExcludeReferenceLink":"",
+  "sysparmInputDisplayValue ":"%s(sysparmInputDisplayValue)"
+}
+
```
Comment (Kanapriya) [inline src/test/resources/artifacts/ESB/config/restRequests/servicenow/putRecordById_optional.json]: Remove unwanted lines

File: src/test/resources/artifacts/ESB/connector/config/servicenow.properties [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+sysparmReadReplicaCategory=""
+KeyValuePairs=active=true
+sysparmInputDisplayValue=true
+
+sysparmAvgFields=category,active
+sysparmMinFields=number
+sysparmMaxFields=number
+sysparmSumFields=priority
+sysparmCount=true
+
```
Comment (Kanapriya) [inline src/test/resources/artifacts/ESB/connector/config/servicenow.properties]: Remove empty space

File: src/test/resources/automation.properties [hunk 10]
```diff
@@ -0,0 +1,166 @@
...
+#gs.service.host.name=gadget.stratosliveqa.private.wso2.com
+#brs.service.host.name=rule.stratosliveqa.private.wso2.com
+#bps.service.host.name=process.stratosliveqa.private.wso2.com
+#manager.service.host.name=stratosliveqa.private.wso2.com
+#ms.service.host.name=mashup.stratosliveqa.private.wso2.com
+#mb.service.host.name=messaging.stratosliveqa.private.wso2.com
+#cep.service.host.name=cep.stratosliveqa.private.wso2.com
+#bam.service.host.name=monitor.stratosliveqa.private.wso2.com
+#is.service.host.name=identity.stratosliveqa.private.wso2.com
+
```
Comment (Kanapriya) [inline src/test/resources/automation.properties]: Remove unwanted lines

File: src/main/assembly/assemble-connector.xml (hunk 1 above)
Comment (yashothara) [inline src/main/assembly/assemble-connector.xml]: Fixed in https://github.com/wso2-extensions/esb-connector-servicenow/pull/9.

File: src/main/assembly/assemble-connector.xml (hunk 2 above)
Comment (yashothara) [inline src/main/assembly/assemble-connector.xml]: Fixed in https://github.com/wso2-extensions/esb-connector-servicenow/pull/9.

File: src/main/assembly/assemble-connector.xml [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+    <includeBaseDirectory>false</includeBaseDirectory>
+
```
Comment (yashothara) [inline src/main/assembly/assemble-connector.xml]: Fixed in https://github.com/wso2-extensions/esb-connector-servicenow/pull/9.

File: src/main/assembly/filter.properties (hunk 3 above)
Comment (yashothara) [inline src/main/assembly/filter.properties]: Fixed in https://github.com/wso2-extensions/esb-connector-servicenow/pull/9.

File: src/main/assembly/filter.properties (hunk 4 above)
Comment (yashothara) [inline src/main/assembly/filter.properties]: Fixed in https://github.com/wso2-extensions/esb-connector-servicenow/pull/9.

File: src/main/resources/table/deleteRecordById.xml [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+  ~
+  ~ Unless required by applicable law or agreed to in writing,
+  ~ software distributed under the License is distributed on an
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  -->
+
+
```
Comment (yashothara) [inline src/main/resources/table/deleteRecordById.xml]: Fixed in https://github.com/wso2-extensions/esb-connector-servicenow/pull/9.

File: src/test/INTEGRATION-TEST.txt (hunk 5 above)
Comment (yashothara) [inline src/test/INTEGRATION-TEST.txt]: Fixed in https://github.com/wso2-extensions/esb-connector-servicenow/pull/9.

File: src/test/resources/artifacts/ESB/config/proxies/servicenow/servicenow.xml [hunk 13]
```diff
@@ -0,0 +1,190 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+   Copyright (c) 2005-2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+
```
Comment (yashothara) [inline src/test/resources/artifacts/ESB/config/proxies/servicenow/servicenow.xml]: Fixed in https://github.com/wso2-extensions/esb-connector-servicenow/pull/9.

File: src/test/resources/artifacts/ESB/config/restRequests/servicenow/patchRecordById_optional.json (hunk 6 above)
Comment (yashothara) [inline src/test/resources/artifacts/ESB/config/restRequests/servicenow/patchRecordById_optional.json]: Fixed in https://github.com/wso2-extensions/esb-connector-servicenow/pull/9.

File: src/test/resources/artifacts/ESB/config/restRequests/servicenow/postRecord_negative.json (hunk 7 above)
Comment (yashothara) [inline src/test/resources/artifacts/ESB/config/restRequests/servicenow/postRecord_negative.json]: Fixed in https://github.com/wso2-extensions/esb-connector-servicenow/pull/9.

File: src/test/resources/artifacts/ESB/config/restRequests/servicenow/putRecordById_optional.json (hunk 8 above)
Comment (yashothara) [inline src/test/resources/artifacts/ESB/config/restRequests/servicenow/putRecordById_optional.json]: Fixed in https://github.com/wso2-extensions/esb-connector-servicenow/pull/9.

File: src/test/resources/artifacts/ESB/connector/config/servicenow.properties (hunk 9 above)
Comment (yashothara) [inline src/test/resources/artifacts/ESB/connector/config/servicenow.properties]: Fixed in https://github.com/wso2-extensions/esb-connector-servicenow/pull/9.

File: src/test/resources/automation.properties (hunk 10 above)
Comment (yashothara) [inline src/test/resources/automation.properties]: Fixed in https://github.com/wso2-extensions/esb-connector-servicenow/pull/9.

**Pull request merged**